### ISSUE TITLE
Async map support and awaitPromises.

### ISSUE THREAD

**Issue by SkeLLLa** [nodefluent/kafka-streams#48]
Hi. 
In most.js there's a method called `.awaitPromises()` that allows `map()` for example to return promises.
The result I want to achieve is:
```
stream
  .from('input')
  .map(async () => {
    return 'value'
  })
  .awaitPromises() 
  .to('output')
```

However it kafka-streams module `.awaitPromises()` not working and throwing error that `TypeError: stream.map(...).awaitPromises is not a function`.

So it there any way to make promises work?

I've tried to use `asyncMap` instead of `map`, but without any success.

*UPD*:
Adding new metod to StreamDSL seems solves the problem and code above work as expected.
```
   awaitPromises(etl){
        this.stream$ = this.stream$.awaitPromises(etl);
        return this;
    }
```
This will also work with filters and other methods that return promises.

**Comment by krystianity** [comment]
Released in kafka-streams@4.2.0 :)

**Comment by SkeLLLa** [comment]
Thanks!
